src/components/include/cos_time.h

```diff
@@ -0,0 +1,50 @@
+/**
+ * Redistribution of this file is permitted under the BSD two clause license.
+ *
+ * Copyright 2018, The George Washington University
+ * Author: Phani Gadepalli, phanikishoreg@gwu.edu
+ */
+
+#ifndef COS_TIME_H
+#define COS_TIME_H
+
+#include <cos_debug.h>
+#include <cos_types.h>
+#include <cos_kernel_api.h>
+
+static inline cycles_t
+time_cyc_per_usec(void)
+{
+	static int cycs = 0;
+
+	if (unlikely(cycs <= 0)) cycs = cos_hw_cycles_per_usec(BOOT_CAPTBL_SELF_INITHW_BASE);
```

Comment:
Add a comment at the top explaining that the hw capability is required.

In the future, we likely want two *variants*, one that uses the hw cap, and one that makes an invocation to get this value (and caches the value in the library).